app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -413,92 +407,82 @@ export default Component.extend(TextareaTextManipulation, {
     return toolbar;
   },
 
-  cachedCookAsync(text) {
-    if (this._cachedCookFunction) {
-      return Promise.resolve(this._cachedCookFunction(text));
-    }
+  async cachedCookAsync(text, options = {}) {
```

Comment:
Unrelated refactoring. Feels like the `async` / `await` reads better/easier than the `Promise`-based version.